src/fmu/dataio/_metadata.py

```diff
@@ -24,6 +38,38 @@
 logger: Final = null_logger(__name__)
 
 
+class JsonSchemaMetadata(BaseModel):
+    """Mixin to inject the $schema field into exported metadata."""
+
+    schema_: AnyHttpUrl = Field(
+        default_factory=lambda: AnyHttpUrl(FmuResultsSchema.url()),
+        alias="$schema",
+        frozen=True,
+    )
+
+
+class ObjectMetadataExport(JsonSchemaMetadata, ObjectMetadata, populate_by_name=True):
+    """Wraps the schema ObjectMetadata, adjusting some values to optional for pragmatic
+    purposes when exporting metadata."""
+
+    # These type ignores are for making the field optional
+    fmu: Optional[fields.FMU]  # type: ignore
+    access: Optional[fields.SsdlAccess]  # type: ignore
+    masterdata: Optional[fields.Masterdata]  # type: ignore
+    # !! Keep UnsetData first in this union
+    data: Union[UnsetData, data.AnyData]  # type: ignore
+    preprocessed: Optional[bool] = Field(alias="_preprocessed", default=None)
```

Comment:
We are no longer creating invalid metadata since version 2.6.0 #793. ➡️  hence we can soon drop the optional fields here:  :boom: 🙂

Reply:
I didn't fully realize this... maybe we can have a chat about it